src/ReportExporter.php

```diff
@@ -36,46 +36,38 @@ class ReportExporter {
 	const DOWNLOAD_EXPORT_ACTION = 'woocommerce_admin_download_report_csv';
 
 	/**
-	 * Email export file download link action.
-	 */
-	const REPORT_EXPORT_EMAIL_ACTION = 'woocommerce_admin_email_report_download_link';
-
-	/**
-	 * Queue instance.
+	 * Get all available scheduling actions.
+	 * Used to determine action hook names and clear events.
 	 *
-	 * @var WC_Queue_Interface
-	 */
-	protected static $queue = null;
-
-	/**
-	 * Get queue instance.
-	 *
-	 * @return WC_Queue_Interface
-	 */
-	public static function queue() {
-		if ( is_null( self::$queue ) ) {
-			self::$queue = WC()->queue();
-		}
-
-		return self::$queue;
+	 * @return array
+	 */
+	public static function get_actions() {
+		return array_merge(
+			self::get_scheduler_actions(),
```

Comment:
Could you elaborate on your confusion here a bit more?  

The idea for `get_actions()` is to allow a scheduler to define its own actions. These actions are used later to help with hooks and action clearing clean-up. The method `get_scheduler_actions()` returns the default actions for a scheduler but any scheduler can opt to not use these and/or add their own. 

>What do you think about moving the actions of a SchedulerTraits consuming class into a parameter to be passed into SchedulerTraits::init() instead?

We could rename this method to `get_default_scheduler_actions()` which may make a bit more sense. But I feel passing an array to another method actually makes this more difficult to follow and does a poorer job of encapsulating actions  in specific schedulers.

This scheduler is missing a `$name` property, which is why these actions names are not fully fleshed out. These should look more like `wc-admin_queue_batches_report_exporter` if the `$name` is set to `report_exporter`. This is a good case for a separate interface to enforce these rules, but I think handling this in a follow-up might be best if we choose to do this.

Reply:
Updated those trait methods to `get_default_scheduler_actions()` in https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/697aed6d461fe221a11d3bc22c9d3648d360692a

Reply:
> The method get_scheduler_actions() returns the default actions for a scheduler but any scheduler can opt to not use these and/or add their own.

Are the default actions not required? Perhaps I'm just not following correctly. If none of the schedulers using the trait merged the default actions in with theirs, would actions and batches still get queued?

If the "default" scheduler actions are always required (for scheduling actions and batches), the interface is brittle if it depends on classes using the trait to preserve them.

Reply:
I see what you mean.

The default actions are not required, but it's likely that a class implementing this trait will use them more often than not.  I think we should allow the option to not use the default actions as other methods within the trait are useful without the batch and scheduler actions this trait provides.

If you think that it would be better to create this as more of an "opt-out" approach for these actions, I can update.

Reply:
Updated this in https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/7b07a4430a7c265c456da3f03c2bff27f58fda1a - let me know what you think.